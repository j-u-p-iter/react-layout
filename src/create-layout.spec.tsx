import * as React from 'react';
import { render } from 'react-testing-library';

import { createLayout } from '.';


describe('create-layout', () => {
  describe('result component', () => {
    it('renders properly with all possible internal parts', () => {
      const headerContentEl = <>Header</>;
      const sidebarContentEl = <>Sidebar</>;
      const footerContentEl = <>Footer</>;
      const contentEl = <>Main Content</>;

      const Layout = createLayout({
        headerContentEl,
        sidebarContentEl,
        footerContentEl,
      }, contentEl);

      const { container } = render(
        <Layout 
          wrapperClassName='wrapper'
          sidebarClassName='sidebar'
          contentClassName='content'
          headerClassName='header'
          footerClassName='footer'
        />
      );
      const layoutEl = container.firstChild;

      expect(layoutEl).toMatchSnapshot();
    });

    it('renders properly without some internal parts and without classNames', () => {
      const sidebarContentEl = <>Sidebar</>;
      const contentEl = <>Main Content</>;

      const Layout = createLayout({
        sidebarContentEl,
      }, contentEl);

      const { container } = render(<Layout />);

      const layoutEl = container.firstChild;

      expect(layoutEl).toMatchSnapshot();
    });
  });

  describe('currying', () => {
    it('works properly', () => {
      const headerContentEl = <>Header</>;
      const sidebarContentEl = <>Sidebar</>;
      const footerContentEl = <>Footer</>;
      const contentEl = <>Main Content</>;

      const createLayoutWithInternalParts = createLayout({
        headerContentEl,
        sidebarContentEl,
        footerContentEl,
      });

      const Layout = createLayoutWithInternalParts(contentEl);

      const { container } = render(<Layout />);

      const layoutEl = container.firstChild;

      expect(layoutEl).toMatchSnapshot();
    })
  });
});
