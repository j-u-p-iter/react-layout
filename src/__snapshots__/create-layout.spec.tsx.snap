// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-layout currying works properly 1`] = `
<div
  class=""
>
  <header
    class=""
  >
    Header
  </header>
  <div
    class=""
  >
    Sidebar
  </div>
  <div
    class=""
  >
    Main Content
  </div>
  <footer
    class=""
  >
    Footer
  </footer>
</div>
`;

exports[`create-layout result component renders properly with all possible internal parts 1`] = `
<div
  class="wrapper"
>
  <header
    class="header"
  >
    Header
  </header>
  <div
    class="sidebar"
  >
    Sidebar
  </div>
  <div
    class="content"
  >
    Main Content
  </div>
  <footer
    class="footer"
  >
    Footer
  </footer>
</div>
`;

exports[`create-layout result component renders properly without some internal parts and without classNames 1`] = `
<div
  class=""
>
  <div
    class=""
  >
    Sidebar
  </div>
  <div
    class=""
  >
    Main Content
  </div>
</div>
`;
